openedx/core/djangoapps/user_authn/views/logout.py

```diff
@@ -55,8 +57,19 @@ def dispatch(self, request, *args, **kwargs):
         # Get the list of authorized clients before we clear the session.
         self.oauth_client_ids = request.session.get(edx_oauth2_provider.constants.AUTHORIZED_CLIENTS_SESSION_KEY, [])
 
+        user = copy.copy(request.user)
         logout(request)
 
+        if not request.user.username:
```

Comment:
Do we need this check?

Reply:
just to make sure the existing user is logged out.